### ISSUE TITLE
virtualenv lost in subprocesses

### ISSUE THREAD

**Issue by mbway** [pypy/pypy#4917]
when running a function with `multiprocessing` on windows, within the subprocess the `sys.executable` and `sys.exec_prefix` point to the interpreter that the active virtualenv was derived from instead of the virtualenv itself. This behaviour is different from cpython.

I have not been able to reproduce the issue on windows 10 locally, but the following github action exhibits the issue:
```yaml
name: TestPyPy

on:
  workflow_dispatch:

jobs:
  test:
    name: Test
    runs-on: windows-latest
    strategy:
      matrix:
        python-version: ['pypy3.10', '3.10']
    steps:
      - uses: actions/checkout@v4
      - uses: actions/setup-python@v5
        with:
          python-version: ${{ matrix.python-version }}
          architecture: "x64"
      - name: Create venv
        run: python -m venv venv
      - name: Run test
        run: venv\Scripts\python.exe example.py
```
where `example.py` is
```python
import multiprocessing
import os
import sys

def worker() -> None:
    print(f"worker process executable: {sys.executable}")
    print(f"worker process exec_prefix: {sys.exec_prefix}")
    print(f"worker process env: {os.environ}")

if __name__ == "__main__":
    print(f"main process executable: {sys.executable}")
    print(f"main process exec_prefix: {sys.exec_prefix}")
    print(f"main process env: {os.environ}")

    p = multiprocessing.Process(target=worker)
    p.start()
    p.join()
```

When run with cpython, both the main process and worker process print the same values but when run with pypy the main process has values:
```
main process executable: D:\a\<repo>\<repo>\venv\Scripts\python.exe
main process exec_prefix: D:\a\<repo>\<repo>\venv
```
but the worker process has values:
```
process executable: C:\hostedtoolcache\windows\PyPy\3.10.13\x86\python.exe
process exec_prefix: C:\hostedtoolcache\windows\PyPy\3.10.13\x86
```

**Comment by devdanzin** [comment]
Thanks for the detailed, clear bug report!

This is the same issue as #4876 (which is a lot less clear), so it should work on PyPy 7.3.14 as it got broken in 7.3.15.
#4893 is intended to fix it.

**Comment by mattip** [comment]
#4893 has been merged into the recent 7.3.16 release. Closing, please reopen or open a new issue if the problem is still there.
